src/System.IO.FileSystem/src/System/IO/Win32FileSystemEnumerable.cs

```diff
@@ -381,19 +387,20 @@ public override bool MoveNext()
         }
 
         [System.Security.SecurityCritical]
-        private SearchResult CreateSearchResult(SearchData localSearchData, Interop.mincore.WIN32_FIND_DATA findData)
+        private bool IsResultIncluded(ref Interop.mincore.WIN32_FIND_DATA findData, out TSource result)
         {
             string findData_fileName = findData.cFileName;
             Contract.Requires(findData_fileName.Length != 0 && !Path.IsPathRooted(findData_fileName),
                 "Expected file system enumeration to not have empty file/directory name and not have rooted name");
 
-            String userPathFinal = Path.Combine(localSearchData.userPath, findData_fileName);
-            String fullPathFinal = Path.Combine(localSearchData.fullPath, findData_fileName);
-            return new SearchResult(fullPathFinal, userPathFinal, findData);
+            string userPathFinal = Path.Combine(_searchData.UserPath, findData_fileName);
+            string fullPathFinal = Path.Combine(_searchData.FullPath, findData_fileName);
```

Comment:
Can these Path.Combines be pushed into the handler's IsResultIncluded such that these strings are only created if, well, the result is included?